openmetadata-ui/src/main/resources/ui/src/pages/DatasetDetailsPage/DatasetDetailsPage.component.tsx

```diff
@@ -577,55 +540,51 @@ const DatasetDetailsPage: FunctionComponent = () => {
     );
   }, [datasetFQN]);
 
-  return (
-    <>
-      {isLoading ? (
-        <Loader />
-      ) : isError ? (
-        <ErrorPlaceHolder>
-          {getEntityMissingError('table', tableFQN)}
-        </ErrorPlaceHolder>
-      ) : (
-        <>
-          {tablePermissions.ViewAll || tablePermissions.ViewBasic ? (
-            <DatasetDetails
-              activeTab={activeTab}
-              columnsUpdateHandler={columnsUpdateHandler}
-              createThread={createThread}
-              dataModel={tableDetails.dataModel}
-              datasetFQN={tableFQN}
-              deletePostHandler={deletePostHandler}
-              descriptionUpdateHandler={descriptionUpdateHandler}
-              entityFieldTaskCount={entityFieldTaskCount}
-              entityFieldThreadCount={entityFieldThreadCount}
-              entityThread={entityThread}
-              feedCount={feedCount}
-              fetchFeedHandler={handleFeedFetchFromFeedList}
-              followTableHandler={followTable}
-              handleExtensionUpdate={handleExtensionUpdate}
-              isEntityThreadLoading={isEntityThreadLoading}
-              isSampleDataLoading={isSampleDataLoading}
-              isTableProfileLoading={isTableProfileLoading}
-              paging={paging}
-              postFeedHandler={postFeedHandler}
-              sampleData={sampleData}
-              setActiveTabHandler={activeTabHandler}
-              settingsUpdateHandler={settingsUpdateHandler}
-              slashedTableName={slashedTableName}
-              tableDetails={tableDetails}
-              tableProfile={tableProfile}
-              tagUpdateHandler={onTagUpdate}
-              unfollowTableHandler={unFollowTable}
-              updateThreadHandler={updateThreadHandler}
-              versionHandler={versionHandler}
-            />
-          ) : (
-            <ErrorPlaceHolder type={ERROR_PLACEHOLDER_TYPE.PERMISSION} />
-          )}
-        </>
-      )}
-    </>
-  );
+  if (isLoading) {
+    return <Loader />;
+  }
+  if (isError) {
+    return (
+      <ErrorPlaceHolder>
+        {getEntityMissingError('table', tableFQN)}
+      </ErrorPlaceHolder>
+    );
+  }
+
+  if (tablePermissions.ViewAll || tablePermissions.ViewBasic) {
```

Comment:
I was expecting it to be other way around, If no permission render ErrorPlaceHolder
Lastly render actual content if all goes well